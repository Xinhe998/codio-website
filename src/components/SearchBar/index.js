import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { FaSearch } from 'react-icons/fa';
import action from '../../actions';

import Button from '../Button';
import TextInput from '../TextInput';
import './index.scss';


const SearchBar = () => {
  const [text, setText] = useState('');
  return (

    <div className="searchBar">
      <TextInput
        className="searchBar_input"
        type="text"
        placeholder="搜尋會員"
        text={text}
        onChange={e => setText(e.target.value)}
        icon={<FaSearch />}
      />
    </div>
  );
};

// SearchBar.propTypes = {

// };

// SearchBar.defaultProps = {

// };

const mapStateToProps = store => ({
  user: store.user,
  editor: store.editor,
});
export default withRouter(
  connect(
    mapStateToProps,
    action,
  )(SearchBar),
);
