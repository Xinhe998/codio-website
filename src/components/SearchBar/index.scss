@import '../../../styles/main.scss';

.searchBar {
  
  .searchBar_input {
    position: relative;
  }

  .searchBar_btn {
    position: absolute;
    top:0;
    right: 0px
  }
}