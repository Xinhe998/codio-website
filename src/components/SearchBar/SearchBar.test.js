/* eslint-disable func-names */
import 'babel-polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow, configure, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchBar from './index';

import fakeImg from '../../assets/smartmockups_jxfuqv8i.jpg';
import fakeAvatar from '../../assets/userImg.png';

import storeFs from '../../reducers';

const store = createStore(storeFs);

// 以該解析器提供給 Enzyme 做渲染 Component 的設置
configure({ adapter: new Adapter() });

describe('SearchBar', function() {
  let testComp;

  beforeEach(() => {
    jest.resetModules();
    this.params = {
      placeholder: 'Test',
    };

    this.makeSubject = () => {
      const { placeholder } = this.params;

      testComp = shallow(
        <Router>
          <Provider store={store}>
            <SearchBar placeholder={placeholder} />
          </Provider>
        </Router>,
      );
      return testComp;
    };
  });
  describe('when it has complete data', () => {
    beforeEach(() => {
      this.subject = this.makeSubject();
    });
    it('should be same as snapshot', async () => {
      const wrapper = render(await testComp);
      expect(wrapper).toMatchSnapshot();
    });
    it('should find className', async () => {
      expect(testComp.find('.searchBar'));
    });
  });
});
