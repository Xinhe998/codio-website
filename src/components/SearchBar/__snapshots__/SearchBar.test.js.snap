// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar when it has complete data should be same as snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "searchBar",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "textinput",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "placeholder": "Test",
              "type": "text",
              "value": "",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "textinput__icon",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "fill": "currentColor",
                    "height": "1em",
                    "stroke": "currentColor",
                    "stroke-width": "0",
                    "viewBox": "0 0 512 512",
                    "width": "1em",
                    "xmlns": "http://www.w3.org/2000/svg",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "d": "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "fill": undefined,
                    "height": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                    "xmlns": "http://www.w3.org/2000/xmlns/",
                  },
                  "x-attribsPrefix": Object {
                    "fill": undefined,
                    "height": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                    "xmlns": "",
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "textinput__icon",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "fill": "currentColor",
                  "height": "1em",
                  "stroke": "currentColor",
                  "stroke-width": "0",
                  "viewBox": "0 0 512 512",
                  "width": "1em",
                  "xmlns": "http://www.w3.org/2000/svg",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "d": "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "fill": undefined,
                  "height": undefined,
                  "stroke": undefined,
                  "stroke-width": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                  "xmlns": "http://www.w3.org/2000/xmlns/",
                },
                "x-attribsPrefix": Object {
                  "fill": undefined,
                  "height": undefined,
                  "stroke": undefined,
                  "stroke-width": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                  "xmlns": "",
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "placeholder": "Test",
                "type": "text",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
